### DISCUSSION TITLE
Wiring help for HyperSPI Rasberry Pi 4 & ESP-WROOM-32

### DISCUSSION THREAD

**Discussion by 1Pete3** [awawa-dev/HyperHDR#639] [Q&A]
Hello,
This is my first project using LEDs and creating circuits so I am a complete noob. I am trying to use HyperSPI since I have sk6812 RGBW LEDs and they keep having random flashes. Currently my set up was working besides that when using sk6812spi. 

The Led strip is behind a 70in TV and both sides of the LED strip are receiving power. The raspberrypi is powered using a USB-C powersupply. I flashed the ESP32 with hyperspi_..._SK6812_RGBW_NEUTRAL.bin

The parts I currently have are :
- [5V 60A Power Supply](https://www.amazon.com/gp/product/B06XK3X3PW/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1)
- [Raspberry Pi 4 Model B - 4 GB RAM](https://www.adafruit.com/product/4296)
- [SN74AHCT125](https://www.amazon.com/dp/B00XW2L39K?psc=1&ref=ppx_yo2ov_dt_b_product_details)
- [ESP-WROOM-32](https://www.amazon.com/dp/B08246MCL5?ref=ppx_yo2ov_dt_b_product_details&th=1)
- [Breadboard](https://www.amazon.com/dp/B07DL13RZH?psc=1&ref=ppx_yo2ov_dt_b_product_details)
- [SK6812 RGBW](https://www.amazon.com/dp/B01N0MA729?ref=ppx_yo2ov_dt_b_product_details&th=1)
- [Capture Card](https://www.amazon.com/gp/product/B09P1FRK7H/ref=ppx_yo_dt_b_asin_title_o05_s01?ie=UTF8&psc=1)
- [HDMI Switch](https://www.amazon.com/gp/product/B07VZZQLTV/ref=ppx_yo_dt_b_asin_title_o08_s00?ie=UTF8&th=1)

How should I power the ESP32, through the pins or is connecting a micro usb from the ESP32 to the USB on the raspberrypi ok?
Is there anything that I am missing?
How would I wire the ESP32 to the Raspberry pi?

![My Setup](https://github.com/awawa-dev/HyperHDR/assets/75858369/d775c645-0a53-4357-8e87-61d0bf908e48)

I attached the parts from the poor diagram. 
[My setup parts.zip](https://github.com/awawa-dev/HyperHDR/files/12608954/My.setup.parts.zip)

**Comment by awawa-dev** [comment]
Hi
You should power the ESP32 via USB only as shown in the diagrams in my last blog post. It is absolutely forbidden to simultaneously supply +5v power via the ESP32 pinout from the LED power supply and simultaneously connect the ESP32 to the USB Raspberry Pi port, which has its own USB-C power supply, as this will end in a serious failure.

**Reply by 1Pete3** [reply]
I was able to get it set up correctly I think
![led set up_bb](https://github.com/awawa-dev/HyperHDR/assets/75858369/26591d19-909a-49c6-ae3c-d5f9ddadc6ad)
 I was having trouble wiring from ESP32 to Raspberry Pi but I looked at your diagram.

**Reply by awawa-dev** [reply]
I will take a closer look later. But you really do not want to short +5v and ground here:
![obraz](https://github.com/awawa-dev/HyperHDR/assets/69086569/0d525749-dcee-4cfc-90ec-bd11f3e9604b)

**Reply by AstaRom** [reply]
Most likely the dark brown jumper should have gone one row higher.
I hope this was done only in this figure, and not in the installation itself before turning it on :)
